File: WKWebView-Print/WKWebView-Print/UIViewController+Print.m
//
//  UIViewController+Print.m
//  WKWebView-Print
//
//  http://github.com/sergemoskalenko/WKWebView-Print-Simple
//  Created by Serge Moskalenko on 19.07.2018.
//  Copyright © 2018 Serge Moskalenko. All rights reserved.
//

#import "UIViewController+Print.h"

@implementation UIViewController(Print)

- (void)printTitle:(NSString *)title webView:(WKWebView *)webView {
    UIPrintInteractionController *controller = [UIPrintInteractionController sharedPrintController];
    void (^completionHandler)(UIPrintInteractionController *, BOOL, NSError *) =
    ^(UIPrintInteractionController *printController, BOOL completed, NSError *error) {
        if(!completed && error){
            NSLog(@"FAILED! due to error in domain %@ with error code %ld",
                  error.domain, (long)error.code);
        }
    };
    UIPrintInfo *printInfo = [UIPrintInfo printInfo];
    printInfo.outputType = UIPrintInfoOutputGeneral;
    printInfo.jobName = title;
    printInfo.duplex = UIPrintInfoDuplexLongEdge;
    controller.printInfo = printInfo;
    controller.showsPageRange = YES;
    
    UIViewPrintFormatter *viewFormatter = webView.viewPrintFormatter;
    viewFormatter.startPage = 0;
    controller.printFormatter = viewFormatter;
    
    if (UI_USER_INTERFACE_IDIOM() == UIUserInterfaceIdiomPad)
        [controller presentFromBarButtonItem:nil animated:YES completionHandler:completionHandler];
    else
        [controller presentAnimated:YES completionHandler:completionHandler];
}
@end

File: WKWebView-Print-Swift/WKWebView-Print-Swift/UIViewController+Print.swift

//
//  UIViewController+Print.swift
//  WKWebView-Print-Swift
//
//  http://github.com/sergemoskalenko/WKWebView-Print-Simple
//  Created by Serge Moskalenko on 19.07.2018.
//  Copyright © 2018 Serge Moskalenko. All rights reserved.
//

import UIKit
import WebKit

extension UIViewController {
    func printTitle(_ title: String?, webView: WKWebView?) {
        let controller = UIPrintInteractionController.shared
        let printCompletionHandler: UIPrintInteractionCompletionHandler = { (controller, success, error) -> Void in
            if success {
                print("Print: OK")
            } else {
                 print("Print: ERROR")
            }
        }

        let printInfo = UIPrintInfo.printInfo()
        printInfo.outputType = .general
        printInfo.jobName = title ?? ""
        printInfo.duplex = .longEdge
        controller.printInfo = printInfo
        controller.showsPageRange = true
        let viewFormatter: UIViewPrintFormatter? = webView?.viewPrintFormatter()
        viewFormatter?.startPage = 0
        controller.printFormatter = viewFormatter
        if UI_USER_INTERFACE_IDIOM() == .pad {
            controller.present(from: self.view.bounds, in: self.view, animated: true, completionHandler: printCompletionHandler)
        } else {
            controller.present(animated: true, completionHandler: printCompletionHandler)
        }
    }
}
